android/engine/src/main/java/org/smartregister/fhircore/engine/rulesengine/RulesFactory.kt

```diff
@@ -450,6 +450,44 @@ constructor(
       }
       return serviceStatus
     }
+
+    fun generateListTaskServiceStatus(tasks: List<Task>): String {
```

Comment:
Why not use this `org.smartregister.fhircore.engine.rulesengine.RulesFactory.RulesEngineService#generateTaskServiceStatus`

Reply:
Can't we use existing functionalities to be able to achieve this?

Reply:
@ellykits  do you have any further comments on this or an alternate method we can use for a list of tasks?

Reply:
@AbdulWahabMemon I agree this method can utilize code from the existing method. Please refactor to reuse what's available

Reply:
@AbdulWahabMemon I agree with @dubdabasoduba and @ellykits  on this. This will reduce the total number of loops greatly and any changes in task service status only need to be changed in a single function and not multiple functions and/or files

Reply:
method updated
@ellykits , @ekigamba , @bennsimon 
cc: @ageryck